src/Translations/MinimalTranslationJobComponentModel.cs

```diff
@@ -55,6 +56,17 @@ public class MinimalTranslationJobComponentModel
         /// </summary>
         public TranslationJobComponentState State { get; set; }
 
+        /// <summary>
+        /// Gets or sets the state via a string name.
+        /// </summary>
+        [JsonIgnore]
+        public string StateName
+        {
+            get => this.State.ToString();
+
+            set => this.State = value.ToEnum<TranslationJobComponentState>();
+        }
```

Comment:
I dont think we need this. State serializes fine to string and if not, the enum should have the Json string serializer attribute added. This seems redundant. If the UI uses this property, let's remove it, and instead use State.